packages/backend/src/launch-vm.ts

```diff
@@ -0,0 +1,185 @@
+import path from 'node:path';
```

Comment:
this file is very specific to Mac with a lot of constants specific to this system, how would this scale with https://github.com/containers/podman-desktop-extension-bootc/issues/878 and https://github.com/containers/podman-desktop-extension-bootc/issues/879

Reply:
One step at a time to be honest! When implementing Linux support, it'll be more modular / the file will be way less specific.

Reply:
I concur to the @axel7083 comment here, it should be designed from the beginning as pluggable runtime, not a list of functions

Like you plug qemu/mac support, then you could plug vfkit/mac, etc.

Having like tons of functions in the same file will make it a pure nightmare to maintain as it's not scoped.
I don't see why architecture needs to be sent to each function, it's something stateful from the runner